### PR TITLE
clusterinfo: Introduce hide tidb and delete etcd key

### PR DESCRIPTION
**Pull request by mapleFU**
This pr allow users hide tidb if it was not alive.

### CODE REVIEW

**Comment by breeswish** [comment]
How is the progress now?

File: ui/src/apps/clusterInfo/RootComponent.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ const App = () => {
                   <ComponentPanel field="pd" data={cluster} />
                 </Col>
               </Row>
-              <ClusterInfoTable cluster={cluster} />
+              <ClusterInfoTable cluster={{cluster, setCluster}} />
```
Comment (baurine) [inline ui/src/apps/clusterInfo/RootComponent.js]: It is better to separate the props likes `<ClusterInfoTable cluster={cluster} onChange={setCluster} />`

File: ui/src/apps/clusterInfo/components/ClusterInfoTable.js [hunk 2]
```diff
@@ -99,4 +122,23 @@ function wrapNode(node, comp, id) {
   }
 }
 
+async function deleteTiDBTopology(node, dataSource, setCluster) {
+  let resp = await client.dashboard.topologyTidbAddressDelete(node.address)
+  if (resp.status === 200) {
```
Comment (baurine) [inline ui/src/apps/clusterInfo/components/ClusterInfoTable.js]: I don't think we should judge `resp.status === 200`, it is already done in the client lib internal, you can consider using `try...catch` to handle the exception.

File: ui/src/apps/clusterInfo/components/ClusterInfoTable.js [hunk 3]
```diff
@@ -31,13 +30,45 @@ function ComponentPanelTable({ cluster }) {
...
+    {
+      title: t('cluster_info.component_table.status'),
+      dataIndex: 'status',
+      key: 'status',
+      width: 100,
+      render: (text, node) => {
+        return (
+          <span>
+            <span>{text} </span>
+            {node !== undefined &&
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/components/ClusterInfoTable.js]: `node && node.status !== 'up'`

File: ui/src/apps/clusterInfo/components/ClusterInfoTable.js [hunk 4]
```diff
@@ -31,13 +30,45 @@ function ComponentPanelTable({ cluster }) {
...
+                <a
+                  onClick={() =>
+                    deleteTiDBTopology(
+                      node,
+                      cluster.cluster,
+                      cluster.setCluster
+                    )
+                  }
+                >
+                  ( {t('cluster_info.component_table.del_db')} )
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/components/ClusterInfoTable.js]: it should be called hide

File: ui/src/apps/clusterInfo/components/ClusterInfoTable.js [hunk 5]
```diff
@@ -99,4 +130,8 @@ function wrapNode(node, comp, id) {
   }
 }
 
+async function deleteTiDBTopology(node, cluster, setCluster) {
+  await client.dashboard.topologyTidbAddressDelete(node.address)
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/components/ClusterInfoTable.js]: You need to do something. e.g. Confirm before hide, letting user know what will happen if xxx, and update the list after hide is succeeded.

File: ui/src/apps/clusterInfo/components/ClusterInfoTable.js [hunk 6]
```diff
@@ -96,7 +137,13 @@ function wrapNode(node, comp, id) {
     version: node.version,
     status_port: node.status_port,
     status: status,
+
+    comp: comp,
   }
 }
 
+async function deleteTiDBTopology(node, cluster, setCluster) {
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/components/ClusterInfoTable.js]: This function not not needed?

File: ui/src/apps/clusterInfo/components/ClusterInfoTable.js [hunk 7]
```diff
@@ -31,13 +32,53 @@ function ComponentPanelTable({ cluster }) {
       key: 'deploy_path',
       ellipsis: true,
     },
+    {
+      title: t('cluster_info.component_table.status'),
+      dataIndex: 'status',
+      key: 'status',
+      width: 100,
+      render: (text, node) => {
+        console.log(node)
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/components/ClusterInfoTable.js]: remove

File: ui/src/apps/clusterInfo/components/ClusterInfoTable.js [hunk 8]
```diff
@@ -1,9 +1,16 @@
-import React from 'react'
-import { Table, Card, Skeleton } from 'antd'
+import React, { useState } from 'react'
+import { Table, Card, Skeleton, Modal } from 'antd'
 import { useTranslation } from 'react-i18next'
 
+import client from '@/utils/client'
+
+const confirm = Modal.confirm;
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/components/ClusterInfoTable.js]: remove

File: ui/src/apps/clusterInfo/components/ClusterInfoTable.js [hunk 9]
```diff
@@ -31,13 +32,53 @@ function ComponentPanelTable({ cluster }) {
...
+            confirm({
+              title: 'hide',
+              content: t('cluster_info.component_table.hide_warning'),
+              onOk() {
+                deleteTiDBTopology(node, cluster.cluster, cluster.setCluster)
+              },
+              onCancel() {},
+            });
+          }
+          console.log("IN BRANCH")
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/components/ClusterInfoTable.js]: remove

File: ui/src/apps/clusterInfo/components/ClusterInfoTable.js [hunk 10]
```diff
@@ -31,13 +32,53 @@ function ComponentPanelTable({ cluster }) {
...
+      title: t('cluster_info.component_table.status'),
+      dataIndex: 'status',
+      key: 'status',
+      width: 100,
+      render: (text, node) => {
+        console.log(node)
+        if (node && node.comp && node.comp === 'tidb') {
+          let showConfirm = () => {
+            confirm({
+              title: 'hide',
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/components/ClusterInfoTable.js]: ???

File: ui/src/apps/clusterInfo/translations/en.yaml [hunk 11]
```diff
@@ -14,6 +14,12 @@ cluster_info:
     address: Address
 
     node_list: NODES LIST
+    hide_db: HIDE
+    action: Action
+
+    hide_warning: Do you want to hide this TiDB key?
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/translations/en.yaml]: What is TiDB key?

File: ui/src/apps/clusterInfo/components/ClusterInfoTable.js [hunk 12]
```diff
@@ -31,9 +30,47 @@ function ComponentPanelTable({ cluster }) {
...
+              content: t('cluster_info.component_table.hide_warning'),
+              onOk() {
+                client.dashboard.topologyTidbAddressDelete(node.address)
+              },
+              onCancel() {},
+            });
+          }
+          return (
+            <span>
+              <span>{text} </span>
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/components/ClusterInfoTable.js]: i18n?

Comment (mapleFU) [reply ui/src/apps/clusterInfo/components/ClusterInfoTable.js]: `{text} ` is already an i18n string when it is used as the argument in `render`

File: ui/src/apps/clusterInfo/translations/en.yaml [hunk 13]
```diff
@@ -14,6 +14,12 @@ cluster_info:
     address: Address
 
     node_list: NODES LIST
+    hide_db: HIDE
+    action: Action
+
+    hide_warning: Do you want to hide this TiDB?
+    hide_confirm: OK
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/translations/en.yaml]: unused?

File: ui/src/apps/clusterInfo/translations/en.yaml [hunk 14]
```diff
@@ -14,6 +14,12 @@ cluster_info:
...
+    hide_cancel: Cancel
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/translations/en.yaml]: unused?

File: ui/src/apps/clusterInfo/components/ClusterInfoTable.js [hunk 15]
```diff
@@ -38,9 +30,47 @@ function ComponentPanelTable({ cluster }) {
...
+    {
+      title: t('cluster_info.component_table.status'),
+      dataIndex: 'status',
+      key: 'status',
+      width: 100,
+      render: (status, node) => {
+        if (node && node.comp && node.comp === 'tidb') {
+          let showConfirm = () => {
+            Modal.confirm({
+              title: t('cluster_info.component_table.hide_db'),
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/components/ClusterInfoTable.js]: Please specify different title. Title !== Action Name

File: ui/src/apps/clusterInfo/components/ClusterInfoTable.js [hunk 16]
```diff
@@ -38,9 +30,47 @@ function ComponentPanelTable({ cluster }) {
...
+              onOk() {
+                client.dashboard.topologyTidbAddressDelete(node.address)
+              },
+              onCancel() {},
+            });
+          }
+          return (
+            <span>
+              <span>{t(`cluster_info.component_table.${status}`)} </span>
+              {node.status && node.status !== 'up' && (
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/components/ClusterInfoTable.js]: what if node is not tidb?

File: ui/src/apps/clusterInfo/components/ClusterInfoTable.js [hunk 17]
```diff
@@ -38,9 +30,47 @@ function ComponentPanelTable({ cluster }) {
...
+                  >
+                    {t('cluster_info.component_table.hide_db')}
+                  </a>
+                  )
+                </span>
+              )}
+            </span>
+          )
+        }
+        return <span>{status}</span>
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/components/ClusterInfoTable.js]: missing i18n?

File: ui/src/apps/clusterInfo/components/ClusterInfoTable.js [hunk 18]
```diff
@@ -38,9 +30,57 @@ function ComponentPanelTable({ cluster }) {
...
+      width: 100,
+      render: (status, node) => {
+        if (!status) {
+          return
+        }
+        let statusBanner = <span>{t(`cluster_info.component_table.${status}`)} </span>;
+        if (node.comp && node.comp === 'tidb') {
+          let showConfirm = () => {
+            Modal.confirm({
+              title: `${t('cluster_info.component_table.hide_db', {
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/components/ClusterInfoTable.js]: hide_db.confirm.title

File: ui/src/apps/clusterInfo/components/ClusterInfoTable.js [hunk 19]
```diff
@@ -38,9 +30,57 @@ function ComponentPanelTable({ cluster }) {
...
+                nodeName: node.address,
+              })}`,
+              content: t('cluster_info.component_table.hide_warning', {
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/components/ClusterInfoTable.js]: hide_db.confirm.content

File: ui/src/apps/clusterInfo/components/ClusterInfoTable.js [hunk 20]
```diff
@@ -38,9 +30,57 @@ function ComponentPanelTable({ cluster }) {
...
+                nodeName: node.address,
+              }),
+              okText: t('cluster_info.component_table.hide_confirm'),
+              cancelText: t('cluster_info.component_table.hide_cancel'),
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/components/ClusterInfoTable.js]: remove

File: ui/src/apps/clusterInfo/components/ClusterInfoTable.js [hunk 21]
```diff
@@ -38,9 +30,57 @@ function ComponentPanelTable({ cluster }) {
...
+              {statusBanner}
+              {node.status && node.status !== 'up' && (
+                <span>
+                  (
+                  <a
+                    onClick={() => {
+                     showConfirm()
+                    }}
+                  >
+                    {t('cluster_info.component_table.hide_db')}
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/components/ClusterInfoTable.js]: hide_db.action

File: ui/src/apps/clusterInfo/components/ClusterInfoTable.js [hunk 22]
```diff
@@ -38,9 +30,57 @@ function ComponentPanelTable({ cluster }) {
...
+    {
+      title: t('cluster_info.component_table.status'),
+      dataIndex: 'status',
+      key: 'status',
+      width: 100,
+      render: (status, node) => {
+        if (!status) {
+          return
+        }
+        let statusBanner = <span>{t(`cluster_info.component_table.${status}`)} </span>;
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/components/ClusterInfoTable.js]: why not 

```js
<span>{t(`cluster_info.component_table.${status}`)} { tidb ? ... : null }</span>;
```

HTML structure is simpler

File: ui/src/apps/clusterInfo/components/ClusterInfoTable.js (hunk 20 above)
Comment (mapleFU) [reply ui/src/apps/clusterInfo/components/ClusterInfoTable.js]: Why...?

File: ui/src/apps/clusterInfo/components/ClusterInfoTable.js (hunk 22 above)
Comment (mapleFU) [reply ui/src/apps/clusterInfo/components/ClusterInfoTable.js]: I think using `statusBanner` is OK, I don't like mixing them together...

File: ui/src/apps/clusterInfo/translations/zh-CN.yaml [hunk 23]
```diff
@@ -14,6 +14,15 @@ cluster_info:
     address: 地址
 
     node_list: 节点列表
+    action: 操作
+
+    hide_db:
+      title: 隐藏 TiDB 节点
+      action: '隐藏 {{nodeName}}'
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/translations/zh-CN.yaml]: what is this?

File: ui/src/apps/clusterInfo/translations/en.yaml [hunk 24]
```diff
@@ -14,6 +14,15 @@ cluster_info:
     address: Address
 
     node_list: NODES LIST
+    action: Action
+
+    hide_db:
+      title: 'HIDE TiDB Node'
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/translations/en.yaml]: Hide

File: ui/src/apps/clusterInfo/translations/en.yaml [hunk 25]
```diff
@@ -14,6 +14,14 @@ cluster_info:
     address: Address
 
     node_list: NODES LIST
+    action: Action
+
+    hide_db:
+      title: HIDE
```
Comment (breeswish) [inline ui/src/apps/clusterInfo/translations/en.yaml]: Hide

**Pull request merged**